hyper/contrib.py

```diff
@@ -43,28 +48,40 @@ def get_connection(self, host, port, scheme, cert=None):
         if cert is not None:
             ssl_context = init_context(cert=cert)
 
+        connection_key = (host, port, scheme, cert, proxy)
```

Comment:
I don't think it resolves the core issue. If you are setting headers for the proxy (e.g. proxy-authorization) then we absolutely must respect that. We need to keep track of them. In 99%+ of cases we will never store anything other than one set of proxy headers and a `None` value, which is pretty minimal IMO.